### DISCUSSION TITLE
How to combine the predicted NER RAW output (in BIOES tags format) into Words?

### DISCUSSION THREAD

**Discussion by karndeepsingh** [explosion/spaCy#10602] [Help: Coding & Implementations] [answered]
Hi,
I have been training a NER model and it was trained with BIOES tags for each word in the sentence. And the model is predicting output word-wise with its NER tags. For example I have a sentence "Alex is going with Marty A.Rick to Los Angeles" and when I pass this sentence to the NER model, I get output in BIOES tag format  as:
```
Alex S-Party_name
is O
going O
with O
Marty B-Party_name
A. I-Party_name
Rick E-Party_name
to O
Los B-Address
Angeles E-Address

```
Here, it is quite visible "Marty" is tagged as "B-Partyname" , "A." is tagged as "I-Party_name", "Rick"  is tagged as "E-Party_name"
similarly, "LOS"  is tagged as "B-Address" and "Angeles" is tagged as "E-Address". I am getting predictions for each word but how to save these predictions as one predicted output like "MARTY A. RICK" as "Party_name" likewise "LOS ANGELES" as "Address". How to save this as a single output, not as an individual word prediction?

Please help to get this output in the desired format. 
Thanks

**Comment by polm** [comment]
You can use `doc.ents` to get each entity as a span, see [the docs](https://spacy.io/usage/linguistic-features#named-entities).

**Comment by adrianeboyd** [comment, marked as answer]
Backing up a step, I think you've accidentally trained your NER model on the individual BIOES tags rather than entity spans with labels like `Party_name` rather than `B-Party_name`. Take another look at how you're converting your training data?

**Reply by karndeepsingh** [reply]
Hi @adrianeboyd @polm,
Thanks for answering. Let me explain you whole scenario, I have trained LayoutLMv2 model and it takes visual as well as text information from an image document and give an output in BIOES tag. And it accepts BIOES tag as a training to do Token-classification. Although, I trained the model and it started predicting out the result in BIOES format but I am unable to combine the results as explained above. Hence, I thought to express my concern here in spacy community for help. 

Please help, on getting final output in combined form not at word level tagging. 
I would be greatful.

thanks

**Reply by adrianeboyd** [reply]
Sorry, I don't understand enough about what your pipeline looks like. What is the "NER model" and which component is adding the named entity annotation to the doc?

The basics are still the same as if you were converting training data: you need to convert this output to labeled `Span` objects and set the entities in the `Doc` object with `doc.ents` or `doc.set_ents()`.

Alternatively, depending on how you're integrating this, you might want to create a `Doc` with `Doc(vocab, words=words, spaces=spaces, ents=ents)` with aligned lists of words/spaces/ents where you've converted the BIOES tags to IOB tags.

**Reply by karndeepsingh** [reply]
> Sorry, I don't understand enough about what your pipeline looks like. What is the "NER model" and which component is adding the named entity annotation to the doc?
> 
> The basics are still the same as if you were converting training data: you need to convert this output to labeled `Span` objects and set the entities in the `Doc` object with `doc.ents` or `doc.set_ents()`.
> 
> Alternatively, depending on how you're integrating this, you might want to create a `Doc` with `Doc(vocab, words=words, spaces=spaces, ents=ents)` with aligned lists of words/spaces/ents where you've converted the BIOES tags to IOB tags.

Thanks @adrianeboyd  Can you provide any sample code to convert the BIOES tagged output to combined word using spacy? That will give me proper understanding.


Thanks again! 
cheers to spacy.

**Reply by adrianeboyd** [reply]
We don't use BIOES anywhere, but you can see how we convert CoNLL 2003 IOB->BILUO->spans:

https://github.com/explosion/spaCy/blob/0d0153db63442ef4523e7e33ac5c863172a61422/spacy/training/converters/conll_ner_to_docs.py#L112-L117
